src/pveclib/vec_int128_ppc.h

```diff
@@ -6848,4 +7005,140 @@ vec_vmsumoud (vui64_t a, vui64_t b, vui128_t c)
+       */
+      if (shb > 0)
+	{
+	  const vui8_t vshl = vec_splat_u8 (shb);
+	  const vui8_t vshr = vec_splat_u8 (8 - shb);
+	  const vui8_t zero = vec_splat_u8 (0);
+	  vui8_t lowbits, highbits;
+
+	  /* Left double shift by 1 octet 'zero' || vrb to isolate the
+	   * high order byte of vrb in to the low 8-bits. Then right
+	   * shift this (8-shb) bits. This provides (128-shb) bits of
+	   * leading '0's. */
+	  lowbits = vec_sld (zero, (vui8_t) vrb, 1);
+	  lowbits = vec_vsrb (lowbits, vshr);
+	  /* Left shift the quadword vra shifting in shb '0' bits.  */
+	  highbits = vec_sll ((vui8_t) vra, vshl);
+	  /* Combines left shifted bits from vra, vrb.  */
+	  result = (vui128_t) vec_or (highbits, lowbits);
+	}
+      else
+	result = vra;
+#endif
+    }
+  else
+    {
+      result = vec_sldqi (vra, vrb, (shb & 7));
+    }
+
+  return ((vui128_t) result);
+}
+
+/** \brief Vector Shift Right Double Quadword by Bit Immediate.
+ *
+ *  Return a vector __int128 that is bits 128-shb:255-shb
+ *  from the (256-bit) double quadword (vra || vrb).
+ *  The shift amount is constant immediate value in the range 0-7.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |   8   | 1 cycle  |
+ *  |power9   |  11   | 1/cycle  |
+ *
+ *  @param vra upper 128-bits of the 256-bit double quadword vector.
+ *  @param vrb lower 128-bits of the 256-bit double quadword vector.
+ *  @param shb Shift amount in the range 0-7.
+ *  @return 128-bits from bits 128-shb:255-shb.
+ */
+static inline vui128_t
+vec_vsrdbi (vui128_t vra, vui128_t vrb, const unsigned int shb)
+{
+  vui128_t result;
+
+  if (__builtin_constant_p (shb) && (shb < 8))
+    {
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+      __asm__(
+	  "vsrdbi %0,%1,%2,%3;\n"
+	  : "=v" (result)
+	  : "v" (vra), "v" (vrb), "K" (shb)
+	  : );
+#else
+      /* For Power7/8/9 the quadword shift left/right instructions
+       * only handle 128-bits.
+       * So shift vra and vrb separately then combine those into
+       * a single 128-bit result.
+       */
+      if (shb > 0)
+	{
+	  const vui8_t vshl = vec_splat_u8 (8 - shb);
+	  const vui8_t vshr = vec_splat_u8 (shb);
+	  const vui8_t zero = vec_splat_u8 (0);
+	  vui8_t lowbits, highbits;
+
+	  /* Left double shift by 15 octet vra || 'zero' to isolate the
+	   * low order byte of vra in to the high 8-bits. Then left
+	   * shift this (8-shb) bits. This provides (128-shb) bits of
+	   * trailing '0's. */
+	  highbits = vec_sld ((vui8_t) vra, zero, 15);
+	  highbits = vec_vslb (highbits, vshl);
+	  /* right shift the quadword vrb shifting in shb '0' bits.  */
+	  lowbits = vec_srl ((vui8_t) vrb, vshr);
+	  /* Combines right shifted bits from vra, vrb.  */
+	  result = (vui128_t) vec_or (highbits, lowbits);
+	}
+      else
+	result = vrb;
+#endif
+    }
+  else
+    {
+#if defined (__clang__) && (__clang_major__ < 6)
+      // A workaround for a constant propagation bug in clang-5
+      if (shb == 0)
+	result = vrb;
```

Comment:
indentation.
was the constant propagation bug reported? maybe put a reference here?

Reply:
No one will accept a bug for Fedora 27 BE Clang-5 (both out of service). They will just tell me to upgrade.

Fixed the indent.